packages/linter/src/generators/convert-to-flat-config/generator.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  formatFiles,
+  getProjects,
+  NxJsonConfiguration,
+  ProjectConfiguration,
+  Tree,
+  updateJson,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+import { ConvertToFlatConfigGeneratorSchema } from './schema';
+import { findEslintFile } from '../utils/eslint-file';
+import { convertEslintJsonToFlatConfig } from './converters/json-converter';
+
+export async function convertToFlatConfigGenerator(
+  tree: Tree,
+  options: ConvertToFlatConfigGeneratorSchema
+) {
+  const eslintFile = findEslintFile(tree);
+  if (!eslintFile) {
+    throw new Error('Could not find root eslint file');
+  }
+  if (!eslintFile.endsWith('.json')) {
+    throw new Error(
+      'Only json eslint config files are supported for conversion'
+    );
+  }
+
+  // rename root eslint config to eslint.config.js
+  convertRootToFlatConfig(tree);
+  // rename and map files
+  const projects = getProjects(tree);
+  for (const [project, projectConfig] of projects) {
+    convertProjectToFlatConfig(tree, project, projectConfig);
+  }
+  // replace references in nx.json
+  updateNxJsonConfig(tree);
+  // install missing packages
+
+  if (!options.skipFormat) {
+    await formatFiles(tree);
+  }
+}
+
+export default convertToFlatConfigGenerator;
+
+function convertRootToFlatConfig(tree: Tree) {
+  if (tree.exists('.eslintrc.base.json')) {
+    convertConfigToFlatConfig(
+      tree,
+      '',
+      '.eslintrc.base.json',
+      'eslint.base.config.js'
+    );
+  }
+  if (tree.exists('.eslintrc.json')) {
+    convertConfigToFlatConfig(tree, '', '.eslintrc.json', 'eslint.config.js');
+  }
+}
+
+function convertProjectToFlatConfig(
```

Comment:
This function needs to take into account `targetDefaults`. Imagine things like:

`projectConfig.targets.lint === {}`, because all of the config lives in nx.json